avm/utilities/pipelines/e2eValidation/resourceRemoval/Initialize-DeploymentRemoval.ps1

```diff
@@ -76,8 +76,10 @@ function Initialize-DeploymentRemoval {
             'Microsoft.Authorization/policyDefinitions'
             'Microsoft.Sql/managedInstances',
             'Microsoft.MachineLearningServices/workspaces',
-            'Microsoft.Resources/resourceGroups',
-            'Microsoft.Compute/virtualMachines'
+            'Microsoft.Compute/virtualMachines',
+            'Microsoft.VirtualMachineImages/imageTemplates', # Must be removed before their MSI
```

Comment:
Ideally, since this is impacting the removal order of all modules, we should trigger validation pipelines for all merged ones :smirk:

Reply:
In general I don't object, but given what this script is actually doing (function-wise) that seems like an overkill. The behavior before was, that it would (try) to remove the imageTemplate randomly before or after the MSI (which is usually removed via the resource group removal). Now that specific resource is removed before - which means no other resource is impacted. So I guess what I'm trying to say is that I would appreciate running tests for all modules only if the behavior for other modules is actually affected (like if I'd change the order if the removal - which would absolutely justify this).

Reply:
We're not modifying the behavior of this module only, but also other modules. I understand your angle, but I'm still for the suggestion. As a general rule, whenever the deployment of other modules is triggered, I'll ask for it to be done before the merge.

Reply:
Can't say I agree to generalize this, but I'll see to get it done and attach the badges. Will keep you posted 💪

Reply:
@eriqua you may not have noticed, but I did that back then ;)